src/pudl/output/eia860.py

```diff
@@ -136,11 +138,168 @@ def plants_eia860(pudl_engine, start_date=None, end_date=None):
+
+        Args:
+            plants: the current plants table to check
+            method_str: A description of the method employed. This will be inserted into
+                the log.
+        """
+        currently_null_len = len(plants[plants.balancing_authority_code_eia.isnull()])
+        logger.info(
+            f"Filled BA codes where {method_str}. "
+            f"Currently {currently_null_len/len(plants):.1%} of records ({currently_null_len}) with no BA codes"
+        )
+
+    plants = make_backfilled_ba_code_column(plants, by_cols=["plant_id_eia"]).pipe(
+        make_consistent_ba_code_column
+    )
+    log_current_ba_code_nulls(plants, method_str="no treatment had been applied")
+    # when the backfilled code and the static code are the same, use either result
+    plants.loc[
+        (
+            plants.balancing_authority_code_eia_bfilled
+            == plants.balancing_authority_code_eia_consistent
+        )
+        & (plants.balancing_authority_code_eia.isnull()),
+        "balancing_authority_code_eia",
+    ] = plants.balancing_authority_code_eia_consistent
+
+    log_current_ba_code_nulls(
+        plants, method_str="backfilling and consistent value is the same"
+    )
+    # we found a pattern of 2013 plants reporting BA codes of PACE when the
+    # consistent option was PACW and the states the plants were located in
+    # are PACW states
+    # use PACW for those records
+    plants.loc[
+        (plants.balancing_authority_code_eia.isnull())
+        & (plants.balancing_authority_code_eia_consistent == "PACW")
+        & (plants.state.isin(["OR", "CA"])),
+        "balancing_authority_code_eia",
+    ] = "PACW"
+    log_current_ba_code_nulls(
+        plants, method_str="consistent code is PACW and state is OR or CA"
+    )
+    # found the opposite! Plants in UT labeled as PACW instead of PACE
+    plants.loc[
+        (plants.balancing_authority_code_eia.isnull())
+        & (plants.balancing_authority_code_eia_consistent == "PACE")
+        & (plants.state.isin(["UT"])),
+        "balancing_authority_code_eia",
+    ] = "PACE"
+    log_current_ba_code_nulls(
+        plants, method_str="consistent code is PACE and state is UT"
+    )
+    # we know SWPP has done a ton of accumulation of smaller BA's
+    plants.loc[
+        (plants.balancing_authority_code_eia.isnull())
+        & (plants.balancing_authority_code_eia_consistent == "SWPP"),
+        "balancing_authority_code_eia",
+    ] = plants.balancing_authority_code_eia_bfilled
+    log_current_ba_code_nulls(
+        plants, method_str="SWPP is most consistent value. Filled w/ oldest BA code"
+    )
+
+    plants.loc[
+        (plants.balancing_authority_code_eia.isnull())
+        & (
+            plants.balancing_authority_code_eia_bfilled
+            != plants.balancing_authority_code_eia_consistent
+        )
+        & (plants.balancing_authority_code_eia_consistent_rate <= 0.8),
```

Comment:
What's the reasoning behind the 80% threshold here? Why not stick with the standard 70% threshold? Can you explain? Does that threshold being reasonable depend on the particular number of years of data we have right now?

Reply:
i responded to this in a previous comment.